flink-cdc-common/src/main/java/org/apache/flink/cdc/common/schema/Selectors.java

```diff
@@ -71,8 +83,20 @@ public boolean isMatch(TableId tableId) {
         }
     }
 
-    /** Match the {@link TableId} against the {@link Selector}s. * */
+    /** Match the {@link TableId} against the {@link Selector}s. */
     public boolean isMatch(TableId tableId) {
+        Boolean cachedResult = cache.getIfPresent(tableId);
+        if (cachedResult != null) {
+            return cachedResult;
+        }
+
+        boolean match = computeIsMatch(tableId);
+        cache.put(tableId, match);
+        return match;
```

Comment:
This could be simplified with `LoadingCache`:

```suggestion
        return cache.getUnchecked(tableId);
```